File: frontend/src/types/report.ts
import { ReportStatuses } from "@/const";

export type Report = {
    id: number;
    title: string;
    status: ReportStatuses;
    responsibleUserId: string;
    pastResponsibleUserId: string;
    creatorUserId: string;
    createdAt: string;
    updatedAt: string;
}

File: backend/bugget-api/Bugget.Entities/BO/ReportBo/Report.cs
namespace Bugget.Entities.BO.ReportBo;

public sealed class Report
{
    public int? Id { get; set; }
    public required string Title { get; set; }
    public int Status { get; set; } = (int)ReportStatus.Backlog;
    public required string ResponsibleUserId { get; set; }
    public required string CreatorUserId { get; set; }
    public DateTimeOffset? CreatedAt { get; set; }
    public DateTimeOffset? UpdatedAt { get; set; }

    public required string[] ParticipantsUserIds { get; set; }
    
    public required Bug[] Bugs { get; set; }
}